app/bundles/LeadBundle/Controller/ListController.php

```diff
@@ -786,9 +787,15 @@ public function viewAction(Request $request, SegmentDependencies $segmentDepende
 
         $permissions = [LeadPermissions::LISTS_CREATE, LeadPermissions::LISTS_VIEW_OWN, LeadPermissions::LISTS_VIEW_OTHER, LeadPermissions::LISTS_EDIT_OWN, LeadPermissions::LISTS_EDIT_OTHER, LeadPermissions::LISTS_DELETE_OWN, LeadPermissions::LISTS_DELETE_OTHER];
 
+        // Audit Log
+        $auditLogModel = $this->getModel('core.auditlog');
+        \assert($auditLogModel instanceof AuditLogModel);
```

Comment:
Please use [action based DI](https://symfony.com/doc/current/controller.html#controller-accessing-services) instead
```suggestion
```

Reply:
I reverted this one cause, the `\Mautic\LeadBundle\Controller\ListController::viewAction` is being called from `\Mautic\LeadBundle\Controller\ListController::createSegmentModifyResponse` LN#401. This caused some errors.

Reply:
Well, I solve this by adding the dependency to the places where the method is being called as well. That's how the other dependencies like $listModel were added too.

Reply:
I updated the PR; Please review it.
The commit. to review https://github.com/mautic/mautic/pull/14411/commits/66e2d1460582cad578d6f709d5471693c1750aaa.